services/basic_types.proto

```diff
@@ -526,314 +770,407 @@ enum TokenKeyValidation {
- * themselves be threshold signatures, to allow complex signature requirements.
- *
- * A Key can be a "key list" where all keys in the list must sign unless specified otherwise in the
- * documentation for a specific transaction type (e.g.  FileDeleteTransactionBody).  Their use is
- * dependent on context. For example, a Hedera file is created with a list of keys, where all of
- * them must sign a transaction to create or modify the file, but only one of them is needed to sign
- * a transaction to delete the file. So it's a single list that sometimes acts as a 1-of-M threshold
- * key, and sometimes acts as an M-of-M threshold key.  A key list is always an M-of-M, unless
- * specified otherwise in documentation. A key list can have nested key lists or threshold keys.
- * Nested key lists are always M-of-M. A key list can have repeated primitive public keys, but all
- * repeated keys are only required to sign once.
- *
- * A Key can contain a ThresholdKey or KeyList, which in turn contain a Key, so this mutual
- * recursion would allow nesting arbitrarily deep. A ThresholdKey which contains a list of primitive
- * keys has 3 levels: ThresholdKey -> KeyList -> Key. A KeyList which contains several primitive
- * keys has 2 levels: KeyList -> Key. A Key with 2 levels of nested ThresholdKeys has 7 levels:
- * Key -> ThresholdKey -> KeyList -> Key -> ThresholdKey -> KeyList -> Key.
- *
- * Each Key should not have more than 46 levels, which implies 15 levels of nested ThresholdKeys.
+ * A Key is an entity representing one or more cryptographic public/private key
+ * pairs and, optionally, the structure for how multiple signatures may be
+ * composed to meet complex multiple-signature authorization requirements.
+ *
+ * A Key can be a public key from either the Ed25519 or ECDSA(secp256k1)
+ * signature schemes. In the ECDSA(secp256k1) case we require the 33-byte
+ * compressed form of the public key. For simplicity, we call these
+ * cryptographic public keys `primitive` keys.<br/>
+ * If an entity has a primitive key associated to it, then the corresponding
+ * private key must sign any transaction to send tokens or perform other
+ * actions requiring authorization.
+ *
+ * A Key can also be the ID of a smart contract instance, which is then
+ * authorized to perform any precompiled contract action that requires
+ * authorization by that key.<br/>
+ * Note that when a key is a smart contract ID, it SHALL NOT imply that the
+ * contract with that ID will actually create a cryptographic signature.
+ * It SHALL signify that when the contract calls a precompiled contract, the
+ * resulting "child transaction" SHALL be implicitly authorized to perform any
+ * action controlled by that key. A contract ID key MAY be "delegatable", which
+ * permits that contract to "authorize" a message frame for which it is
+ * recipient, even if the _code_ within that message frame is from a different
+ * contract.
```

Comment:
```suggestion
 * A Key can also be the ID of a smart contract, which authorizes that contract
 * to execute any system contract whose signing requirements include the key. 
 * For example, if account `0.0.A` has a threshold key whose threshold is satisfied
 * by a contract ID key for contract `0.0.C`, then when `0.0.C` is called, it will be
 * authorized to use system contracts to manage any asset owned by `0.0.A`. If the
 * contract ID key is "delegatable", then `0.0.C` can even perform these actions 
 * when running code accessed via `DELEGATECALL`.
```